base/boot.jl

```diff
@@ -465,6 +465,12 @@ struct InitError <: WrappedException
     error
 end
 
+struct ABIOverwrite
```

Comment:
Minor bikeshed, but I prefer `ABIOverride`

ABIOverwrite makes me think that there was some "transient" state before the ABI was "overwritten"